### PR TITLE
Update to use the new API for custom allocators.

### PR DESCRIPTION
**Pull request by too-r**
Had to change the linked_list_allocator dependecy to link directly to git, because the version on crates.io is still using the old `Unique` API.

### CODE REVIEW

**Comment by 4e554c4c** [comment]
Why are you creating `Cargo.lock`?

**Comment by too-r** [comment]
@4e554c4c Cargo.lock is included in Git by default with this repository?

**Comment by 4e554c4c** [comment]
Then add it to the .gitignore. There isn't currently a `Cargo.lock` in the repository so you shouldn't add one without a good reason.

**Comment by too-r** [comment]
Ok, fixed that.

**Comment by too-r** [comment]
It seems I can't use Git apparently, `Cargo.lock` is properly removed now.

**Comment by 4e554c4c** [comment]
Thank you! I think this is good.

File: libs/hole_list_allocator/src/lib.rs [hunk 1]
```diff
@@ -7,62 +7,51 @@
...
-        Heap::new(HEAP_START, HEAP_SIZE)
-    });
-}
+static HEAP: Mutex<Option<Heap>> = Mutex::new(None);
 
-#[no_mangle]
-pub extern fn __rust_allocate(size: usize, align: usize) -> *mut u8 {
-    HEAP.lock().allocate_first_fit(size, align).expect("out of memory")
+//Set up the heap
+pub unsafe fn init(offset: usize, size: usize) {
```
Comment (phil-opp) [inline libs/hole_list_allocator/src/lib.rs]: This init function isn't called right now, so the heap stays uninitialized.

**Comment by too-r** [comment]
I think using the LockedHeap struct is a definite possiblity, however for now I have just changed it so the heap init function does get called in `src/memory/mod.rs` under the `memory::init` function.

**Comment by phil-opp** [comment]
Ok, I decided to merge this for now. Maybe we switch to LockedHeap eventually, but for now it's good to have this project buildable again.

Thanks a lot @too-r!

File: src/lib.rs [hunk 2]
```diff
@@ -53,7 +53,7 @@ pub extern "C" fn rust_main(multiboot_information_address: usize) {
 
     // initialize our IDT
     interrupts::init(&mut memory_controller);
-
+
```
Comment (bjorn3) [inline src/lib.rs]: trailing whitespace which was not previously here.

**Pull request merged**